src/compute-types/src/plan/lowering.rs

```diff
@@ -479,10 +479,9 @@ impl Context {
                                 // to check this here for robustness against future code changes.
                                 break 'fusion None;
                             }
-                            mfp.permute(
-                                (1..mfp.input_arity).map(|col| (col, col - 1)).collect(),
-                                mfp.input_arity - 1,
-                            );
+                            let permutation: BTreeMap<_, _> =
+                                (1..mfp.input_arity).map(|col| (col, col - 1)).collect();
+                            mfp.permute_fn(|c| permutation[&c], mfp.input_arity - 1);
```

Comment:
We could write this as:
```suggestion
                            mfp.permute_fn(|c| c - 1, mfp.input_arity - 1);
```
but it has different behavior when accessing undefined columns.

Reply:
Sanity check, but .. this feels unrelated to the PR? Or .. I'm pretty sure this isn't related to `(key, val)` to `row` mapping, just an incidental use of `permute_fn`?